pydoctor/test/test_packages.py

```diff
@@ -47,3 +47,6 @@ def test_allgames() -> None:
     # InSourceAll is not moved into mod2, but NotInSourceAll is.
     assert 'InSourceAll' in system.allobjects['allgames.mod1'].contents
     assert 'NotInSourceAll' in system.allobjects['allgames.mod2'].contents
+    moved = system.allobjects['allgames.mod2'].contents['NotInSourceAll']
```

Comment:
It would be nice to have a docstring. I find it hard to read the test.

I see that the test data is at https://github.com/twisted/pydoctor/blob/master/pydoctor/test/testpackages/allgames/mod2.py

and that for pydoctor mod2 is considered as being sourced from mod1.py

Reply:
I've added some explanations in the test case.

The new checks basically verify the "when a function or class is reparented, we now report the original source file (where the docstring resides) instead of the new parent's source file" item from the PR summary.